packages/core/src/modules/credentials/repository/CredentialExchangeRecord.ts

```diff
@@ -22,7 +22,7 @@ export interface CredentialExchangeRecordProps {
   createdAt?: Date
   state: CredentialState
   connectionId?: string
-  threadId: string
+  threadId?: string
```

Comment:
Making this optional has some downsides. What if we create the threadId beforehand for the message?

So:
1. create the message (without any attachments, just empty proposal message)
2. create the credential record (with the threadId from the previous step)
3. call the format services and add the attachment formats to the message

Or:
1. generate an uuid
2. create the credential record with threadId set to uuid from previous step
3. call the format services and get the attachment formats
4. create the proposal message, passing the uuid from step 1 as the message id

My preference would go to the first appraoch, but if for some reason that's not possible the second appraoch still has my preference to making this property optional